Source/WebKit/UIProcess/API/C/WKPageConfigurationRef.cpp

```diff
@@ -67,65 +67,66 @@ void WKPageConfigurationSetPageGroup(WKPageConfigurationRef, WKPageGroupRef)
 
 WKUserContentControllerRef WKPageConfigurationGetUserContentController(WKPageConfigurationRef configuration)
 {
-    return toAPI(toImpl(configuration)->userContentController());
+    return toAPI(toProtectedImpl(configuration)->protectedUserContentController().get());
 }
 
 void WKPageConfigurationSetUserContentController(WKPageConfigurationRef configuration, WKUserContentControllerRef userContentController)
 {
-    toImpl(configuration)->setUserContentController(toImpl(userContentController));
+    toProtectedImpl(configuration)->setUserContentController(toProtectedImpl(userContentController));
 }
 
 WKPreferencesRef WKPageConfigurationGetPreferences(WKPageConfigurationRef configuration)
 {
-    return toAPI(toImpl(configuration)->preferences());
+    return toAPI(toProtectedImpl(configuration)->protectedPreferences().get());
 }
 
 void WKPageConfigurationSetPreferences(WKPageConfigurationRef configuration, WKPreferencesRef preferences)
 {
-    toImpl(configuration)->setPreferences(toImpl(preferences));
+    toProtectedImpl(configuration)->setPreferences(toProtectedImpl(preferences));
 }
 
 WKPageRef WKPageConfigurationGetRelatedPage(WKPageConfigurationRef configuration)
 {
-    return toAPI(toImpl(configuration)->relatedPage());
+    RefPtr relatedPage = toProtectedImpl(configuration)->relatedPage();
```

Comment:
Please add a `protectedRelatedPage()` getter instead of introducing an unnecessary local variable.